### PR TITLE
refactor(x): refactor assign tag to post to use services [Start #117]

### PR DESCRIPTION
**Pull request by beblicarl**
#### Pre-Submission Checklist

-   [x] Your pull request targets the `main` branch.
-   [x] Branch name is descriptive and follows the guidelines [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices).
-   [x] All tests pass . Use `git commit --amend` to amend any fixes.

#### Checklist:

-   [x] All tests are passing.
-   [] Tested changes locally.
-   [x] Improved documentation or readme where necessary.
-   [x] Your commit messages format follows our commit message guideline [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices#commits).

#### What does this PR do?
This PR refactors assign tag to post to use services

#### How should this be manually tested?

1. Start server with `npm run dev`
2. You will notice the logic of the assign tag to post endpoint uses services

#### What are the relevant issues?
The plan was to refactor the assign tag to post so it uses services

### CODE REVIEW

File: src/services/posts.js [hunk 1]
```diff
@@ -88,12 +89,30 @@ const queryPostTags = async({tag}) => {
...
+		// eslint-disable-next-line max-len
+		`INSERT INTO posts_tags ("postId","tagId") 
+		SELECT $1,$2 WHERE NOT EXISTS 
+		(SELECT * FROM posts_tags WHERE "postId" = $1 AND "tagId" = $2) 
+		RETURNING *`,
+		[postId, tagId]
+	)
+	const postsTags = result.rows[0]
+	if (!postsTags) {
+		const errorMessage = TagAlreadyAssignedToPostError.message
```
Comment (jocrah) [inline src/services/posts.js:103]: @beblicarl this way of assigning errors a bit verbose and I think we're doing it in more than 5 places now so let's refactor this into a function

1. create a `custom-error.js` file in the `src/lib` folder
2. create a function called `customError` which takes `name` and `message` as parameters and now does this logic there and export it. something like

```js
const customError = ({name, message}) => {
  const error = new Error(message)
  err.name =  name
  return err
}
```
4.  import the `customError` here and use that instead

so your service code will now look like
```js
throw customError(TagAlreadyAssignedToPostError)
```

File: src/routes/articles.js [hunk 2]
```diff
@@ -4,14 +4,16 @@ const { logger } = require('../lib')
...
+	TagAlreadyAssignedToPostError
 } = require("../services/errors")
 
 const log = logger()
 const router = express.Router()
 const ERROR_MAP = {
 	[ArticleDoesNotExistForCommentError.name] : 422 ,
-	[ArticleDoesNotExistError.name] : 404
+	[ArticleDoesNotExistError.name] : 404,
+	[TagAlreadyAssignedToPostError] : 400
```
Comment (jocrah) [inline src/routes/articles.js:16]: @beblicarl, 
```suggestion
	[TagAlreadyAssignedToPostError.name] : 400
```

also let's make it `422`. The client request is correct (not necessarily malformed) but the server cannot process it due to the constraint.

Comment (beblicarl) [reply src/routes/articles.js:16]: oh okay, lemme make the change then

Comment (beblicarl) [reply src/routes/articles.js]: done @jocrah

**Review by jocrah — APPROVED** [review]

**Pull request merged**